test/config_tests.ml

```diff
@@ -0,0 +1,124 @@
+let pmsg =
+  Alcotest.testable (fun ppf (`Msg s) -> Fmt.pf ppf "Error @[<v>(%s)@]" s)
+    (=)
+
+let result_p_err (type x) ok =
+  (* like Alcotest.result, but pretty-prints the Error `Msg and
+     fails the test when [Error _] is encountered.
+     On success, [ok]'s [pp] and [equal]
+     function are used to pretty-print and determine test result. *)
+  let module Ok = (val ok : Alcotest.TESTABLE with type t = x) in
+  let pp = (fun ppf -> function
+      | Ok v -> Fmt.pf ppf "Ok @[<v>(%a)@]" Ok.pp v
+      | Error `Msg v -> Fmt.pf ppf "Error @[<v>(%s)@]" v) in
+  Alcotest.testable pp
+    (fun a b -> match a,b with
+       | Ok a, Ok b -> Ok.equal a b
+       | _ -> false)
+
+let conf_map = Alcotest.testable
+    Openvpn.Config.pp Openvpn.Config.(equal eq)
+
+let parse_noextern conf =
+  Openvpn.Config.parse_client ~string_of_file:(fun path ->
+      Rresult.R.error_msgf
+        "this test suite does not read external files, \
+         but a config asked for: %S" path) conf
+
+
+let ok_minimal_client () =
+  (* verify that we can parse a minimal good config. *)
+  let basic =
+    {|tls-client
+    auth-user-pass [inline]
+    <auth-user-pass>
+testuser
+testpass
+</auth-user-pass>
+remote 10.0.0.1|} in
+  Alcotest.(check (result pass pmsg)) "basic conf works"
+    (Ok Openvpn.Config.empty)
```

Comment:
I don't understand why `Openvpn.Config.empty` is the result of this config test -- I'd expect there being at least remote and the auth-user-pass keys and values being present.

Reply:
Because I was being lazy and put `Alcotest.pass` on the `ok`-side of `Alcotest.result`, so I just needed "a value."

I think a cleaner design would be to have a global `Config.t` that represented this minimal config, since it looks like we end up repeating it quite a lot since all tests need `client`, `remote`, and a way to authenticate atm. I will move things around a bit.

Reply:
implemented that now by copy-pasting over. I didn't like the idea of exposing it in `openvpn.mli`, since it lacks some keys (`remote`, auth, ...) and thus doesn't pass validation without additions.

On that note, wonder if we should expose the `is_valid_client_config` function.